client/post-editor/editor-post-formats/index.jsx

```diff
@@ -35,6 +36,18 @@ const EditorPostFormats = React.createClass( {
 		};
 	},
 
+	getSelectedPostFormat: function() {
+		var validSlugs = this.getPostFormats().map( function( postFormat ) {
```

Comment:
A few points to make here:

Should we shortcut the return if the selected value is `standard`, since we know (based on `getPostFormats`) that this option will always be available, and is very likely to be the default post format?

We should use `const` and `let` in all new code ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#variable-declarations)).

We could further optimize to eliminate a few cycles:

```js
const isSupportedFormat = this.getPostFormats().some( ( postFormat ) => {
	return postFormat.slug === this.props.value;
} );

if ( isValidSlug ) {
	return this.props.value;
}

return 'standard';
```

Benefit of using [`Array.prototype.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is that it stops execution when it finds a matching result.